### PR TITLE
mp_image: pass rotation correctly to/from AVFrame

### PR DESCRIPTION
**Pull request by kasper93**
Fixes rotating image by --video-rotate when filtering.

Fixes: #12771

### CODE REVIEW

File: video/mp_image.c [hunk 1]
```diff
@@ -1044,7 +1044,6 @@ struct mp_image *mp_image_from_av_frame(struct AVFrame *src)
 
     if (src->opaque_ref) {
         struct mp_image_params *p = (void *)src->opaque_ref->data;
-        dst->params.rotate = p->rotate;
```
Comment (sfan5) [inline video/mp_image.c:1047]: why was this removed?

File: video/mp_image.c [hunk 2]
```diff
@@ -1197,6 +1196,12 @@ struct AVFrame *mp_image_to_av_frame(struct mp_image *src)
         clm->MaxCLL = src->params.color.sig_peak * MP_REF_WHITE;
     }
 
+    AVFrameSideData * rot_sd = av_frame_new_side_data(dst,
+                                                      AV_FRAME_DATA_DISPLAYMATRIX,
+                                                      sizeof(int32_t) * 9);
+    if (rot_sd)
+        av_display_rotation_set((int32_t *)rot_sd->data, src->params.rotate);
```
Comment (sfan5) [inline video/mp_image.c:1203]:
```suggestion
    {
        AVFrameSideData *sd = av_frame_new_side_data(dst,
                                                     AV_FRAME_DATA_DISPLAYMATRIX,
                                                     sizeof(int32_t) * 9);
        MP_HANDLE_OOM(sd);
        av_display_rotation_set((int32_t *)sd->data, src->params.rotate);
    }
```

Comment (kasper93) [reply video/mp_image.c:1203]: > could optionally add if (src->params.rotate)

We also want to set if it is 0 to support for example `--video-rotate=no` which is set on the output from decoder, so we would lost this information.

Comment (sfan5) [reply video/mp_image.c:1203]: a rotation of 0 is a no-op so I don't see how you could possibly lose any information
either way my main suggestion wasn't about adding a condition

Comment (kasper93) [reply video/mp_image.c:1203]: > a rotation of 0 is a no-op so I don't see how you could possibly lose any information

Exactly and we have to preserve this information when `--video-rotate=no` is used, because `AV_FRAME_DATA_DISPLAYMATRIX` may have different rotation set.

The flow during filtering is

1. lavc decode
2. avframe->mp_image (read AV_FRAME_DATA_DISPLAYMATRIX)
4. fix_image_params (apply `--video-rotate` and container rotate if available and AV_FRAME_DATA_DISPLAYMATRIX was not available)
5. mp_image->avframe (set/replace adjusted rotation is side data)
6. filter
7. avframe->mp_image (read AV_FRAME_DATA_DISPLAYMATRIX)
8. display

>  either way my main suggestion wasn't about adding a condition

I know, I can update the in the evening.

Comment (sfan5) [reply video/mp_image.c:1203]: Ah I see, it copies existing side data right below this.

Comment (kasper93) [reply video/mp_image.c]: doen

**Review by sfan5 — APPROVED** [review]
